// sidsc-components/esm2022/dsc-tab-group/dsc-tab/dsc-tab.component.mjs
import { Component, Input, ViewChild, ViewEncapsulation } from '@angular/core';
import { MatTab, MatTabsModule } from '@angular/material/tabs';
import { MatIconModule } from '@angular/material/icon';
import { NgIf, NgClass, NgFor, NgTemplateOutlet, NgComponentOutlet, CommonModule } from '@angular/common';
import { coerceBooleanProperty } from '@angular/cdk/coercion';
import { DscBadgeDirective } from 'sidsc-components/dsc-badge';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/tabs";
import * as i2 from "@angular/material/icon";
export class DscTabComponent {
    constructor() {
        this.label = '';
        this.iconStyle = 'filled';
        this.dscBadgeSize = 'standard';
        this._disabled = false;
    }
    get disabled() {
        return this._disabled;
    }
    set disabled(value) {
        this._disabled = coerceBooleanProperty(value);
    }
    getIconClass() {
        if (this.iconStyle === 'filled')
            return 'material-icons';
        if (this.iconStyle === 'rounded')
            return 'material-icons-round';
        return `material-icons-${this.iconStyle}`;
    }
    isImage(icon) {
        if (typeof icon !== 'string')
            return false;
        const lower = icon.toLowerCase();
        return lower.endsWith('.svg') || lower.endsWith('.png');
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscTabComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DscTabComponent, isStandalone: true, selector: "dsc-tab", inputs: { label: "label", icon: "icon", iconStyle: "iconStyle", dscBadge: "dscBadge", dscBadgeSize: "dscBadgeSize", disabled: "disabled" }, viewQueries: [{ propertyName: "matTab", first: true, predicate: MatTab, descendants: true }], ngImport: i0, template: "<mat-tab [disabled]=\"disabled\">\n  <ng-template mat-tab-label>\n    <mat-icon *ngIf=\"icon\" [ngClass]=\"getIconClass()\">\n      <img *ngIf=\"isImage(icon)\" src=\"{{icon}}\" alt=\"\">\n      {{ icon }}\n    </mat-icon>\n    {{ label }}\n    <div class=\"badge\" [ngClass]=\"{\n      'badge-standard': dscBadgeSize === 'standard',\n      'badge-other': dscBadgeSize !== 'standard'\n    }\"\n     *ngIf=\"dscBadge\" [dscBadge]=\"dscBadge\" [dscBadgeSize]=\"dscBadgeSize\"></div>\n  </ng-template>\n  <ng-content></ng-content>\n</mat-tab>\n", styles: [".badge-standard{right:-12px;top:24px}.badge-other{right:-12px;align-items:center}\n"], dependencies: [{ kind: "ngmodule", type: MatTabsModule }, { kind: "directive", type: i1.MatTabLabel, selector: "[mat-tab-label], [matTabLabel]" }, { kind: "component", type: i1.MatTab, selector: "mat-tab", inputs: ["disabled"], exportAs: ["matTab"] }, { kind: "directive", type: NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "ngmodule", type: CommonModule }, { kind: "ngmodule", type: MatIconModule }, { kind: "component", type: i2.MatIcon, selector: "mat-icon", inputs: ["color", "inline", "svgIcon", "fontSet", "fontIcon"], exportAs: ["matIcon"] }, { kind: "directive", type: DscBadgeDirective, selector: "[dscBadge]", inputs: ["dscBadgeDisabled", "dscBadgeSize", "dscBadgeVariant", "dscBadgeVariantDark", "dscBadgeOverlap", "dscBadgePosition", "dscBadge", "dscBadgeDescription", "dscBadgeHidden"] }], encapsulation: i0.ViewEncapsulation.None }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscTabComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dsc-tab', standalone: true, imports: [MatTabsModule, NgClass, NgFor, NgIf, NgTemplateOutlet, NgComponentOutlet, CommonModule, MatIconModule, DscBadgeDirective], encapsulation: ViewEncapsulation.None, template: "<mat-tab [disabled]=\"disabled\">\n  <ng-template mat-tab-label>\n    <mat-icon *ngIf=\"icon\" [ngClass]=\"getIconClass()\">\n      <img *ngIf=\"isImage(icon)\" src=\"{{icon}}\" alt=\"\">\n      {{ icon }}\n    </mat-icon>\n    {{ label }}\n    <div class=\"badge\" [ngClass]=\"{\n      'badge-standard': dscBadgeSize === 'standard',\n      'badge-other': dscBadgeSize !== 'standard'\n    }\"\n     *ngIf=\"dscBadge\" [dscBadge]=\"dscBadge\" [dscBadgeSize]=\"dscBadgeSize\"></div>\n  </ng-template>\n  <ng-content></ng-content>\n</mat-tab>\n", styles: [".badge-standard{right:-12px;top:24px}.badge-other{right:-12px;align-items:center}\n"] }]
        }], propDecorators: { matTab: [{
                type: ViewChild,
                args: [MatTab]
            }], label: [{
                type: Input
            }], icon: [{
                type: Input
            }], iconStyle: [{
                type: Input
            }], dscBadge: [{
                type: Input
            }], dscBadgeSize: [{
                type: Input
            }], disabled: [{
                type: Input
            }] } });
