// sidsc-components/esm2022/dsc-snackbar/dsc-snackbar.module.mjs
import { NgModule } from '@angular/core';
import { MatSnackBarModule } from '@angular/material/snack-bar';
import { MatIconModule } from '@angular/material/icon';
import { NgClass, NgIf } from '@angular/common';
import { DscButtonComponent } from 'sidsc-components/dsc-button';
import { DscSnackbarComponent } from './dsc-snackbar.component';
import { DscSnackbarService } from './shared/dsc-snackbar.service';
import * as i0 from "@angular/core";
export class DscSnackbarModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscSnackbarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: DscSnackbarModule, declarations: [DscSnackbarComponent], imports: [MatSnackBarModule, MatIconModule, NgClass, NgIf, DscButtonComponent], exports: [DscSnackbarComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscSnackbarModule, providers: [DscSnackbarService], imports: [MatSnackBarModule, MatIconModule, DscButtonComponent] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscSnackbarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DscSnackbarComponent],
                    imports: [MatSnackBarModule, MatIconModule, NgClass, NgIf, DscButtonComponent],
                    exports: [DscSnackbarComponent],
                    providers: [DscSnackbarService]
                }]
        }] });
