// sidsc-components/esm2022/core/services/breakpoint-matcher.service.mjs
import { Injectable } from '@angular/core';
import { Breakpoints } from '@angular/cdk/layout';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/cdk/layout";
export class BreakpointMatcherService {
    constructor(_breakpointObserver) {
        this._breakpointObserver = _breakpointObserver;
        this._breakpoints = {
            xs: Breakpoints.XSmall,
            sm: Breakpoints.Small,
            md: Breakpoints.Medium,
            lg: Breakpoints.Large,
            xl: Breakpoints.XLarge
        };
    }
    getSize$() {
        return this._breakpointObserver.observe(Object.values(this._breakpoints)).pipe(map(result => {
            const matchingBreakpoints = Object.keys(this._breakpoints).filter(
            // @ts-ignore
            i => result.breakpoints[this._breakpoints[i]]);
            return matchingBreakpoints[0] || 'unknown';
        }));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BreakpointMatcherService, deps: [{ token: i1.BreakpointObserver }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BreakpointMatcherService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BreakpointMatcherService, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: function () { return [{ type: i1.BreakpointObserver }]; } });
