// sidsc-components/esm2022/core/services/field-error-messenger.service.mjs
import { Injectable } from '@angular/core';
import { formatCurrency } from '@angular/common';
import * as i0 from "@angular/core";
export class FieldErrorMessengerService {
    getErrorMessage(formField, customErrorMessage) {
        if (!formField)
            return customErrorMessage || '';
        const errorMessages = {
            required: 'Campo obrigatório.',
            email: 'Email inválido.',
            pattern: 'Formato inválido.',
            minlength: (error) => `O tamanho mínimo é ${error.requiredLength}.`,
            maxlength: (error) => `O tamanho máximo é ${error.requiredLength}.`,
            min: (error) => `O valor mínimo é ${formatCurrency(error.min, 'pt-br', 'R$')}.`,
            max: (error) => `O valor máximo é ${formatCurrency(error.max, 'pt-br', 'R$')}.`,
            matDatepickerFilter: 'Data não permitida.',
            matDatepickerMin: 'Data inferior ao mínimo.',
            matDatepickerMax: 'Data superior ao máximo.',
            notAllowedFile: 'Formato de arquivo inválido.',
            fileSizeOverLimit: (error) => `O tamanho máximo é ${error.maxSizeFriendly}.`,
        };
        for (const error in formField.errors) {
            const message = errorMessages[error];
            if (message) {
                return typeof message === 'function' ? message(formField.errors[error]) : message;
            }
        }
        return customErrorMessage || '';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: FieldErrorMessengerService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: FieldErrorMessengerService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: FieldErrorMessengerService, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }] });
