// sidsc-components/esm2022/core/base-control-value-accessor.mjs
import { coerceBooleanProperty } from '@angular/cdk/coercion';
export class BaseControlValueAccessor {
    constructor() {
        this._disabled = false;
        this.onChange = (_) => { };
        this.onTouched = () => { };
    }
    get val() {
        return this._val;
    }
    set val(value) {
        if (this._val !== value) {
            this._val = value;
            if (this.updateOn === 'change') {
                this.onChange(value);
                this.onTouched();
            }
        }
    }
    get disabled() {
        return this._disabled;
    }
    set disabled(value) {
        this._disabled = coerceBooleanProperty(value);
    }
    get updateOn() {
        return this._updateOn ?? 'change';
    }
    set updateOn(value) {
        this._updateOn = value;
    }
    writeValue(value) {
        this.val = value;
    }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    setDisabledState(value) {
        this.disabled = value;
    }
    emitValueOn() {
        this.onChange(this._val);
        this.onTouched();
    }
}
