// sidsc-components/esm2022/core/base-component.component.mjs
import { Component, EventEmitter, inject, Input, LOCALE_ID, Output } from '@angular/core';
import { registerLocaleData } from '@angular/common';
import localePt from '@angular/common/locales/pt';
import { FieldErrorMessengerService } from './services/field-error-messenger.service';
import { BaseControlValueAccessor } from './base-control-value-accessor';
import { CustomErrorStateMatcher } from './custom-error-state-matcher';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
registerLocaleData(localePt, 'pt');
export class BaseComponent extends BaseControlValueAccessor {
    get type() {
        return this._type;
    }
    set type(value) {
        this.isPasswordField = value === 'password';
        this._type = value;
    }
    get size() {
        return this._size;
    }
    set size(value) {
        this._size = value;
        this.formFieldLabelSize = `mat-label--${value}`;
        this.formFieldSize = `mat-mdc-form-field--${value}`;
        this.selectOptionSize = `mat-mdc-select-panel--${value}`;
        this.selectOption = `mat-mdc-option--${value}`;
    }
    get maxlength() {
        return this._maxlength;
    }
    set maxlength(value) {
        const length = typeof value === 'string' ? parseInt(value) : value;
        if (!Number.isNaN(length) && Number.isInteger(length))
            this._maxlength = length;
    }
    constructor(_ngControl, _parentForm, _parentFormGroup) {
        super();
        this._ngControl = _ngControl;
        this._parentForm = _parentForm;
        this._parentFormGroup = _parentFormGroup;
        this.errorMessageService = inject(FieldErrorMessengerService);
        this.blur = new EventEmitter();
        this.formFieldHint = '';
        this.showIconPassword = true;
        this.readOnly = false;
        this._type = 'text';
        this.isPasswordField = false;
        this._size = 'standard';
        this.formFieldLabelSize = 'mat-label--standard';
        this.formFieldSize = 'mat-mdc-form-field--standard';
        this.selectOptionSize = 'mat-mdc-select-panel--standard';
        this.selectOption = 'mat-mdc-option--standard';
        this._maxlength = 45;
        this.matcher = new CustomErrorStateMatcher(this._ngControl, this._parentForm, this._parentFormGroup);
    }
    get errorText() {
        return this.errorMessageService.getErrorMessage(this._ngControl, this.errorMessage);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BaseComponent, deps: [{ token: i1.NgControl }, { token: i1.NgForm }, { token: i1.FormGroupDirective }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: BaseComponent, isStandalone: true, selector: "ng-component", inputs: { label: "label", labelHint: "labelHint", labelHintTooltip: "labelHintTooltip", name: "name", placeholder: "placeholder", formFieldHint: "formFieldHint", errorMessage: "errorMessage", showIconPassword: "showIconPassword", readOnly: "readOnly", ariaDescribedby: ["aria-describedby", "ariaDescribedby"], ariaLabel: ["aria-label", "ariaLabel"], ariaLabelledby: ["aria-labelledby", "ariaLabelledby"], type: "type", size: "size", maxlength: "maxlength" }, outputs: { blur: "blur" }, providers: [{ provide: LOCALE_ID, useValue: 'pt' }], usesInheritance: true, ngImport: i0, template: '', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: BaseComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                    standalone: true,
                    providers: [{ provide: LOCALE_ID, useValue: 'pt' }]
                }]
        }], ctorParameters: function () { return [{ type: i1.NgControl }, { type: i1.NgForm }, { type: i1.FormGroupDirective }]; }, propDecorators: { blur: [{
                type: Output
            }], label: [{
                type: Input
            }], labelHint: [{
                type: Input
            }], labelHintTooltip: [{
                type: Input
            }], name: [{
                type: Input
            }], placeholder: [{
                type: Input
            }], formFieldHint: [{
                type: Input
            }], errorMessage: [{
                type: Input
            }], showIconPassword: [{
                type: Input
            }], readOnly: [{
                type: Input
            }], ariaDescribedby: [{
                type: Input,
                args: ['aria-describedby']
            }], ariaLabel: [{
                type: Input,
                args: ['aria-label']
            }], ariaLabelledby: [{
                type: Input,
                args: ['aria-labelledby']
            }], type: [{
                type: Input
            }], size: [{
                type: Input
            }], maxlength: [{
                type: Input
            }] } });
