// sidsc-components/esm2022/dsc-dialog/dsc-dialog.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule, NgClass, NgIf, NgTemplateOutlet } from '@angular/common';
import { MatIconModule } from '@angular/material/icon';
import { MatDialogModule } from '@angular/material/dialog';
import { DscButtonComponent } from 'sidsc-components/dsc-button';
import { DscDialogComponent } from './dsc-dialog.component';
import { DscDialogService } from './shared/dsc-dialog.service';
import { DscTooltipModule } from 'sidsc-components/dsc-tooltip';
import * as i0 from "@angular/core";
export class DscDialogModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscDialogModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: DscDialogModule, declarations: [DscDialogComponent], imports: [MatDialogModule,
            NgClass,
            NgIf,
            DscButtonComponent,
            DscTooltipModule,
            MatIconModule,
            NgTemplateOutlet,
            CommonModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscDialogModule, providers: [DscDialogService], imports: [MatDialogModule,
            DscButtonComponent,
            DscTooltipModule,
            MatIconModule,
            CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscDialogModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DscDialogComponent],
                    imports: [
                        MatDialogModule,
                        NgClass,
                        NgIf,
                        DscButtonComponent,
                        DscTooltipModule,
                        MatIconModule,
                        NgTemplateOutlet,
                        CommonModule
                    ],
                    providers: [DscDialogService]
                }]
        }] });
