// sidsc-components/esm2022/dsc-stepper/shared/dsc-stepper-button.mjs
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/cdk/stepper";
export class CdkStepperNext {
    constructor(_stepper) {
        this._stepper = _stepper;
        /** Type of the next button. Defaults to "submit" if not specified. */
        this.type = 'submit';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: CdkStepperNext, deps: [{ token: i1.CdkStepper }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.12", type: CdkStepperNext, isStandalone: true, selector: "dsc-button[cdkStepperNext]", inputs: { type: "type" }, host: { listeners: { "click": "_stepper.next()" }, properties: { "type": "type" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: CdkStepperNext, decorators: [{
            type: Directive,
            args: [{
                    selector: 'dsc-button[cdkStepperNext]',
                    standalone: true,
                    host: {
                        '[type]': 'type',
                        '(click)': '_stepper.next()'
                    }
                }]
        }], ctorParameters: function () { return [{ type: i1.CdkStepper }]; }, propDecorators: { type: [{
                type: Input
            }] } });
/** Button that moves to the previous step in a stepper workflow. */
export class CdkStepperPrevious {
    constructor(_stepper) {
        this._stepper = _stepper;
        /** Type of the previous button. Defaults to "button" if not specified. */
        this.type = 'button';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: CdkStepperPrevious, deps: [{ token: i1.CdkStepper }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.12", type: CdkStepperPrevious, isStandalone: true, selector: "dsc-button[cdkStepperPrevious]", inputs: { type: "type" }, host: { listeners: { "click": "_stepper.previous()" }, properties: { "type": "type" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: CdkStepperPrevious, decorators: [{
            type: Directive,
            args: [{
                    selector: 'dsc-button[cdkStepperPrevious]',
                    standalone: true,
                    host: {
                        '[type]': 'type',
                        '(click)': '_stepper.previous()'
                    }
                }]
        }], ctorParameters: function () { return [{ type: i1.CdkStepper }]; }, propDecorators: { type: [{
                type: Input
            }] } });
export class CdkStepperReset {
    constructor(_stepper) {
        this._stepper = _stepper;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: CdkStepperReset, deps: [{ token: i1.CdkStepper }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.12", type: CdkStepperReset, isStandalone: true, selector: "dsc-button[cdkStepperReset]", host: { listeners: { "click": "_stepper.reset()" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: CdkStepperReset, decorators: [{
            type: Directive,
            args: [{
                    selector: 'dsc-button[cdkStepperReset]',
                    standalone: true,
                    host: {
                        '(click)': '_stepper.reset()'
                    }
                }]
        }], ctorParameters: function () { return [{ type: i1.CdkStepper }]; } });
export class DscStepperNext extends CdkStepperNext {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscStepperNext, deps: null, target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.12", type: DscStepperNext, isStandalone: true, selector: "dsc-button[dscStepperNext]", inputs: { type: "type" }, host: { properties: { "type": "type" }, classAttribute: "mat-stepper-next" }, usesInheritance: true, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscStepperNext, decorators: [{
            type: Directive,
            args: [{
                    selector: 'dsc-button[dscStepperNext]',
                    host: {
                        'class': 'mat-stepper-next',
                        '[type]': 'type'
                    },
                    standalone: true,
                    inputs: ['type']
                }]
        }] });
/** Button that moves to the previous step in a stepper workflow. */
export class DscStepperPrevious extends CdkStepperPrevious {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscStepperPrevious, deps: null, target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.12", type: DscStepperPrevious, isStandalone: true, selector: "dsc-button[dscStepperPrevious]", inputs: { type: "type" }, host: { properties: { "type": "type" }, classAttribute: "mat-stepper-previous" }, usesInheritance: true, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscStepperPrevious, decorators: [{
            type: Directive,
            args: [{
                    selector: 'dsc-button[dscStepperPrevious]',
                    host: {
                        'class': 'mat-stepper-previous',
                        '[type]': 'type'
                    },
                    standalone: true,
                    inputs: ['type']
                }]
        }] });
export class DscStepperReset extends CdkStepperReset {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscStepperReset, deps: null, target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.12", type: DscStepperReset, isStandalone: true, selector: "dsc-button[dscStepperReset]", usesInheritance: true, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscStepperReset, decorators: [{
            type: Directive,
            args: [{
                    standalone: true,
                    selector: 'dsc-button[dscStepperReset]'
                }]
        }] });
