// sidsc-components/esm2022/dsc-dialog/shared/dsc-dialog.service.mjs
import { Injectable } from '@angular/core';
import { take } from 'rxjs';
import { DscDialogComponent } from '../dsc-dialog.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/dialog";
export class DscDialogService {
    constructor(dialog) {
        this.dialog = dialog;
    }
    confirm(dscDialogConfig) {
        const dialogRef = this.dialog.open(DscDialogComponent, {
            disableClose: dscDialogConfig.disableClose,
            autoFocus: dscDialogConfig.autoFocus,
            restoreFocus: dscDialogConfig.restoreFocus,
            hasBackdrop: true,
            ariaLabel: dscDialogConfig.ariaLabel,
            ariaLabelledBy: dscDialogConfig.ariaLabelledBy,
            ariaDescribedBy: dscDialogConfig.ariaDescribedBy,
            data: dscDialogConfig.data
        });
        dialogRef.afterClosed().pipe(take(1)).subscribe(value => {
            const actionButton = dscDialogConfig.data?.actionButton;
            if (value && actionButton?.confirmFunction) {
                actionButton.confirmFunction(value);
            }
            else if (!value && actionButton?.cancelFunction) {
                actionButton.cancelFunction();
            }
        });
        return dialogRef;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscDialogService, deps: [{ token: i1.MatDialog }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscDialogService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscDialogService, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: function () { return [{ type: i1.MatDialog }]; } });
