// sidsc-components/esm2022/dsc-sidenav/dsc-sidenav.component.mjs
import { Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
import { MatSidenavModule } from '@angular/material/sidenav';
import { MatListModule } from '@angular/material/list';
import { NgClass, NgForOf, NgIf } from '@angular/common';
import { coerceBooleanProperty, coerceNumberProperty } from '@angular/cdk/coercion';
import { FormControl, ReactiveFormsModule } from '@angular/forms';
import { debounceTime, distinctUntilChanged, map, takeUntil } from 'rxjs/operators';
import { Subject } from 'rxjs';
import { DscInputComponent } from 'sidsc-components/dsc-input';
import { isEmptyArray, isNotEmptyArray, replaceAccents } from 'sidsc-components/core';
import { DscSidenavItemComponent } from './dsc-sidenav-item/dsc-sidenav-item.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "@angular/material/sidenav";
import * as i3 from "@angular/material/list";
import * as i4 from "@angular/forms";
export class DscSidenavComponent {
    constructor(_router) {
        this._router = _router;
        this._onDestroy = new Subject();
        this.expandedItems = new Map();
        this.singleExpand = true;
        this._opened = true;
        this._fixedInViewport = false;
        this._fixedTopGap = 0;
        this._fixedBottomGap = 0;
        this._mode = 'side';
        this._showFilter = false;
        this.openedChange = new EventEmitter();
        this.onClick = new EventEmitter();
        this.search = new FormControl('');
    }
    get menu() {
        return this._menu;
    }
    set menu(value) {
        this._menu = value ? value : [];
        this.menuFiltered = this._menu;
    }
    get opened() {
        return this._opened;
    }
    set opened(value) {
        this._opened = coerceBooleanProperty(value);
    }
    get fixedInViewport() {
        return this._fixedInViewport;
    }
    set fixedInViewport(value) {
        this._fixedInViewport = coerceBooleanProperty(value);
    }
    get fixedTopGap() {
        return this._fixedTopGap;
    }
    set fixedTopGap(value) {
        this._fixedTopGap = coerceNumberProperty(value);
    }
    get fixedBottomGap() {
        return this._fixedBottomGap;
    }
    set fixedBottomGap(value) {
        this._fixedBottomGap = coerceNumberProperty(value);
    }
    get mode() {
        return this._mode;
    }
    set mode(value) {
        this._mode = value;
    }
    get showFilter() {
        return this._showFilter;
    }
    set showFilter(value) {
        this._showFilter = coerceBooleanProperty(value);
    }
    ngOnInit() {
        this._filterInit();
        setTimeout(() => this._applyDefaultSelection(), 0);
    }
    onExpand(menu) {
        if (this.singleExpand) {
            this.expandedItems.forEach((_, key) => this.expandedItems.set(key, false));
        }
        this.expandedItems.set(menu.title, true);
    }
    _applyDefaultSelection() {
        if (!this.menu || !this.menu.length)
            return;
        const path = this._findPath(this.menu, m => !!m.defaultSelected);
        if (!path || !path.length)
            return;
        const target = path[path.length - 1];
        path.slice(0, -1).forEach(parent => this.expandedItems.set(parent.title, true));
        const parentOfTarget = path.length > 1 ? path[path.length - 2] : undefined;
        if (parentOfTarget)
            this.onExpand(parentOfTarget);
        if (target.url) {
            const current = this._router.url.split('#')[0];
            if (current !== target.url)
                this._router.navigateByUrl(target.url);
        }
        this.onClick.emit(target);
    }
    onMenuClick(menu) {
        this.onClick.emit(menu);
    }
    _findPath(nodes, predicate) {
        for (const n of nodes) {
            if (predicate(n))
                return [n];
            if (n.children && n.children.length) {
                const childPath = this._findPath(n.children, predicate);
                if (childPath.length)
                    return [n, ...childPath];
            }
        }
        return [];
    }
    isExpanded(menu) {
        return this.expandedItems.get(menu.title) || false;
    }
    _filterInit() {
        this.search.valueChanges
            .pipe(map(text => (text && text.length > 3 ? text : '')), distinctUntilChanged(), debounceTime(450), takeUntil(this._onDestroy))
            .subscribe(value => {
            const menuAux = JSON.parse(JSON.stringify(this.menu));
            this.menuFiltered = value ? this._doFilter(menuAux, value) : menuAux;
        });
    }
    _doFilter(menuList, value) {
        let menuListFiltered = [];
        menuList.forEach((menu) => {
            if (!menu.disabled) {
                if (isEmptyArray(menu.children)) {
                    if (this._compare(menu.title, value))
                        menuListFiltered.push(menu);
                }
                else {
                    const collapsibleMenu = JSON.parse(JSON.stringify(menu));
                    collapsibleMenu.children = this._doFilter(collapsibleMenu.children, value);
                    collapsibleMenu.expanded = true;
                    if (isNotEmptyArray(collapsibleMenu.children))
                        menuListFiltered.push(collapsibleMenu);
                }
            }
        });
        return menuListFiltered;
    }
    _compare(s1, s2) {
        return replaceAccents(s1).trim().toLowerCase().includes(replaceAccents(s2).trim().toLowerCase());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscSidenavComponent, deps: [{ token: i1.Router }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.2.12", type: DscSidenavComponent, isStandalone: true, selector: "dsc-sidenav", inputs: { customClass: "customClass", coverSrc: "coverSrc", singleExpand: "singleExpand", menu: "menu", opened: "opened", fixedInViewport: "fixedInViewport", fixedTopGap: "fixedTopGap", fixedBottomGap: "fixedBottomGap", mode: "mode", showFilter: "showFilter" }, outputs: { openedChange: "openedChange", onClick: "onClick" }, ngImport: i0, template: "<mat-sidenav-container [ngClass]=\"customClass\">\n  <mat-sidenav class=\"mat-elevation-z2\"\n               [mode]=\"mode\"\n               [opened]=\"opened\"\n               [fixedInViewport]=\"fixedInViewport\"\n               [fixedTopGap]=\"fixedTopGap\"\n               [fixedBottomGap]=\"fixedBottomGap\"\n               (openedChange)=\"this.openedChange.emit($event)\">\n    <div class=\"mat-sidenav-content\">\n      <div class=\"mat-sidenav-header\"\n            *ngIf=\"coverSrc || showFilter\">\n        <div class=\"mat-sidenav-header__cover\"\n             *ngIf=\"coverSrc\">\n          <img [src]=\"coverSrc\" alt=\"\" tabindex=\"-1\">\n        </div>\n        <div class=\"mat-sidenav-header__filter\"\n             *ngIf=\"showFilter\">\n          <dsc-input placeholder=\"Pesquisar\"\n                     iconPrefix=\"search\"\n                     [formControl]=\"search\">\n          </dsc-input>\n        </div>\n      </div>\n      <mat-nav-list role=\"navigation\">\n        <dsc-sidenav-item *ngFor=\"let menu of menuFiltered\"\n                          [menu]=\"menu\"\n                          [expanded]=\"isExpanded(menu)\"\n                          (expandChange)=\"onExpand(menu)\"\n                          (onClick)=\"onMenuClick($event)\">\n        </dsc-sidenav-item>\n      </mat-nav-list>\n    </div>\n  </mat-sidenav>\n  <mat-sidenav-content role=\"main\">\n    <ng-content></ng-content>\n  </mat-sidenav-content>\n</mat-sidenav-container>\n", styles: ["dsc-sidenav .mat-sidenav-container{--mat-sidenav-container-divider-color: var(--dsc-color-bg-neutral-1);--mat-sidenav-container-background-color: var(--dsc-color-bg-neutral-1);--mat-sidenav-content-background-color: var(--dsc-color-bg-neutral-1)}dsc-sidenav .mat-sidenav-container .mat-sidenav{width:260px;box-shadow:4px 0 12px #00000029;border:none}dsc-sidenav .mat-sidenav-container .mat-sidenav-content .mat-sidenav-header{display:flex;flex-direction:column;gap:var(--dsc-spacing-micro);margin:var(--dsc-spacing-micro)}dsc-sidenav .mat-sidenav-container .mat-sidenav-content .mat-sidenav-header__cover img{max-width:236px;max-height:236px}dsc-sidenav .mat-sidenav-container .mat-sidenav-content .mat-sidenav-header__filter .mat-mdc-form-field-subscript-wrapper{height:0}dsc-sidenav .mat-sidenav-container .mat-sidenav-content .mat-mdc-nav-list{display:flex;flex-direction:column;padding:0}dsc-sidenav .mat-sidenav-container .mat-sidenav .mat-drawer-inner-container{overflow-x:hidden}dsc-sidenav .mat-sidenav-container .mat-sidenav .mat-drawer-inner-container::-webkit-scrollbar{display:none}dsc-sidenav .mat-sidenav-container .mat-sidenav-content{background:var(--dsc-color-bg-neutral-1)}\n"], dependencies: [{ kind: "ngmodule", type: MatSidenavModule }, { kind: "component", type: i2.MatSidenav, selector: "mat-sidenav", inputs: ["fixedInViewport", "fixedTopGap", "fixedBottomGap"], exportAs: ["matSidenav"] }, { kind: "component", type: i2.MatSidenavContainer, selector: "mat-sidenav-container", exportAs: ["matSidenavContainer"] }, { kind: "component", type: i2.MatSidenavContent, selector: "mat-sidenav-content" }, { kind: "ngmodule", type: MatListModule }, { kind: "component", type: i3.MatNavList, selector: "mat-nav-list", exportAs: ["matNavList"] }, { kind: "component", type: DscSidenavItemComponent, selector: "dsc-sidenav-item", inputs: ["menu", "depth", "expanded", "selectedKey"], outputs: ["onClick", "expandChange", "selectedKeyChange"] }, { kind: "directive", type: NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "component", type: DscInputComponent, selector: "dsc-input", inputs: ["suffix", "mask", "iconPrefix", "iconSuffix", "iconStyle", "showCharCounter", "showClearButton"], outputs: ["keydown"] }, { kind: "ngmodule", type: ReactiveFormsModule }, { kind: "directive", type: i4.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i4.FormControlDirective, selector: "[formControl]", inputs: ["formControl", "disabled", "ngModel"], outputs: ["ngModelChange"], exportAs: ["ngForm"] }, { kind: "directive", type: NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], encapsulation: i0.ViewEncapsulation.None }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscSidenavComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dsc-sidenav', standalone: true, encapsulation: ViewEncapsulation.None, imports: [
                        MatSidenavModule,
                        MatListModule,
                        DscSidenavItemComponent,
                        NgForOf,
                        NgClass,
                        DscInputComponent,
                        ReactiveFormsModule,
                        NgIf
                    ], template: "<mat-sidenav-container [ngClass]=\"customClass\">\n  <mat-sidenav class=\"mat-elevation-z2\"\n               [mode]=\"mode\"\n               [opened]=\"opened\"\n               [fixedInViewport]=\"fixedInViewport\"\n               [fixedTopGap]=\"fixedTopGap\"\n               [fixedBottomGap]=\"fixedBottomGap\"\n               (openedChange)=\"this.openedChange.emit($event)\">\n    <div class=\"mat-sidenav-content\">\n      <div class=\"mat-sidenav-header\"\n            *ngIf=\"coverSrc || showFilter\">\n        <div class=\"mat-sidenav-header__cover\"\n             *ngIf=\"coverSrc\">\n          <img [src]=\"coverSrc\" alt=\"\" tabindex=\"-1\">\n        </div>\n        <div class=\"mat-sidenav-header__filter\"\n             *ngIf=\"showFilter\">\n          <dsc-input placeholder=\"Pesquisar\"\n                     iconPrefix=\"search\"\n                     [formControl]=\"search\">\n          </dsc-input>\n        </div>\n      </div>\n      <mat-nav-list role=\"navigation\">\n        <dsc-sidenav-item *ngFor=\"let menu of menuFiltered\"\n                          [menu]=\"menu\"\n                          [expanded]=\"isExpanded(menu)\"\n                          (expandChange)=\"onExpand(menu)\"\n                          (onClick)=\"onMenuClick($event)\">\n        </dsc-sidenav-item>\n      </mat-nav-list>\n    </div>\n  </mat-sidenav>\n  <mat-sidenav-content role=\"main\">\n    <ng-content></ng-content>\n  </mat-sidenav-content>\n</mat-sidenav-container>\n", styles: ["dsc-sidenav .mat-sidenav-container{--mat-sidenav-container-divider-color: var(--dsc-color-bg-neutral-1);--mat-sidenav-container-background-color: var(--dsc-color-bg-neutral-1);--mat-sidenav-content-background-color: var(--dsc-color-bg-neutral-1)}dsc-sidenav .mat-sidenav-container .mat-sidenav{width:260px;box-shadow:4px 0 12px #00000029;border:none}dsc-sidenav .mat-sidenav-container .mat-sidenav-content .mat-sidenav-header{display:flex;flex-direction:column;gap:var(--dsc-spacing-micro);margin:var(--dsc-spacing-micro)}dsc-sidenav .mat-sidenav-container .mat-sidenav-content .mat-sidenav-header__cover img{max-width:236px;max-height:236px}dsc-sidenav .mat-sidenav-container .mat-sidenav-content .mat-sidenav-header__filter .mat-mdc-form-field-subscript-wrapper{height:0}dsc-sidenav .mat-sidenav-container .mat-sidenav-content .mat-mdc-nav-list{display:flex;flex-direction:column;padding:0}dsc-sidenav .mat-sidenav-container .mat-sidenav .mat-drawer-inner-container{overflow-x:hidden}dsc-sidenav .mat-sidenav-container .mat-sidenav .mat-drawer-inner-container::-webkit-scrollbar{display:none}dsc-sidenav .mat-sidenav-container .mat-sidenav-content{background:var(--dsc-color-bg-neutral-1)}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.Router }]; }, propDecorators: { customClass: [{
                type: Input
            }], coverSrc: [{
                type: Input
            }], singleExpand: [{
                type: Input
            }], menu: [{
                type: Input
            }], opened: [{
                type: Input
            }], fixedInViewport: [{
                type: Input
            }], fixedTopGap: [{
                type: Input
            }], fixedBottomGap: [{
                type: Input
            }], mode: [{
                type: Input
            }], showFilter: [{
                type: Input
            }], openedChange: [{
                type: Output
            }], onClick: [{
                type: Output
            }] } });
