// sidsc-components/esm2022/dsc-snackbar/shared/dsc-snackbar.service.mjs
import { Injectable } from '@angular/core';
import { DscSnackbarComponent } from '../dsc-snackbar.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/snack-bar";
export class DscSnackbarService {
    constructor(snackBar) {
        this.snackBar = snackBar;
    }
    add(snackbarConfig) {
        this.snackBar.openFromComponent(DscSnackbarComponent, {
            verticalPosition: snackbarConfig.verticalPosition ?? 'bottom',
            horizontalPosition: snackbarConfig.horizontalPosition ?? 'center',
            duration: snackbarConfig.duration ?? 5000,
            data: snackbarConfig
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscSnackbarService, deps: [{ token: i1.MatSnackBar }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscSnackbarService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscSnackbarService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.MatSnackBar }]; } });
