// sidsc-components/esm2022/dsc-tooltip/dsc-tooltip.directive.mjs
import { Directive, Inject, Input, Optional } from '@angular/core';
import { _MatTooltipBase, MAT_TOOLTIP_DEFAULT_OPTIONS, MAT_TOOLTIP_SCROLL_STRATEGY } from '@angular/material/tooltip';
import { DOCUMENT } from '@angular/common';
import { DscTooltipComponent } from './dsc-tooltip.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/cdk/overlay";
import * as i2 from "@angular/cdk/platform";
import * as i3 from "@angular/cdk/a11y";
import * as i4 from "@angular/cdk/bidi";
export class DscTooltipDirective extends _MatTooltipBase {
    get placement() {
        return this._placement;
    }
    set placement(value) {
        this._placement = value ?? 'above';
        super.position = this._placement;
    }
    get variant() {
        return this._variant;
    }
    set variant(value) {
        this._variant = value;
        super.tooltipClass = {
            'mat-tooltip--highlight': this.variant === 'highlight'
        };
    }
    get positionAtOrigin() {
        return super.positionAtOrigin;
    }
    set positionAtOrigin(value) {
        super.positionAtOrigin = value;
    }
    get disabled() {
        return super.disabled;
    }
    set disabled(value) {
        super.disabled = value;
    }
    get showDelay() {
        return super.showDelay;
    }
    set showDelay(value) {
        super.showDelay = value;
    }
    get hideDelay() {
        return super.hideDelay;
    }
    set hideDelay(value) {
        super.hideDelay = value;
    }
    get message() {
        return super.message;
    }
    set message(value) {
        super.message = value;
    }
    constructor(overlay, elementRef, scrollDispatcher, viewContainerRef, ngZone, platform, ariaDescriber, focusMonitor, scrollStrategy, dir, defaultOptions, _document) {
        super(overlay, elementRef, scrollDispatcher, viewContainerRef, ngZone, platform, ariaDescriber, focusMonitor, scrollStrategy, dir, defaultOptions, _document);
        this._tooltipComponent = DscTooltipComponent;
        this._cssClassPrefix = 'mat-mdc';
        this._variant = 'neutral';
        this._viewportMargin = 8;
        super.position = 'above';
    }
    ngAfterViewInit() {
        super.ngAfterViewInit();
    }
    ngOnDestroy() {
        super.ngOnDestroy();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscTooltipDirective, deps: [{ token: i1.Overlay }, { token: i0.ElementRef }, { token: i1.ScrollDispatcher }, { token: i0.ViewContainerRef }, { token: i0.NgZone }, { token: i2.Platform }, { token: i3.AriaDescriber }, { token: i3.FocusMonitor }, { token: MAT_TOOLTIP_SCROLL_STRATEGY }, { token: i4.Directionality, optional: true }, { token: MAT_TOOLTIP_DEFAULT_OPTIONS, optional: true }, { token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.2.12", type: DscTooltipDirective, selector: "[dscTooltip]", inputs: { placement: ["dscTooltipPosition", "placement"], variant: ["dscTooltipVariant", "variant"], positionAtOrigin: ["dscTooltipPositionAtOrigin", "positionAtOrigin"], disabled: ["dscTooltipDisabled", "disabled"], showDelay: ["dscTooltipShowDelay", "showDelay"], hideDelay: ["dscTooltipHideDelay", "hideDelay"], message: ["dscTooltip", "message"] }, usesInheritance: true, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscTooltipDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[dscTooltip]'
                }]
        }], ctorParameters: function () { return [{ type: i1.Overlay }, { type: i0.ElementRef }, { type: i1.ScrollDispatcher }, { type: i0.ViewContainerRef }, { type: i0.NgZone }, { type: i2.Platform }, { type: i3.AriaDescriber }, { type: i3.FocusMonitor }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [MAT_TOOLTIP_SCROLL_STRATEGY]
                }] }, { type: i4.Directionality, decorators: [{
                    type: Optional
                }] }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [MAT_TOOLTIP_DEFAULT_OPTIONS]
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }]; }, propDecorators: { placement: [{
                type: Input,
                args: ['dscTooltipPosition']
            }], variant: [{
                type: Input,
                args: ['dscTooltipVariant']
            }], positionAtOrigin: [{
                type: Input,
                args: ['dscTooltipPositionAtOrigin']
            }], disabled: [{
                type: Input,
                args: ['dscTooltipDisabled']
            }], showDelay: [{
                type: Input,
                args: ['dscTooltipShowDelay']
            }], hideDelay: [{
                type: Input,
                args: ['dscTooltipHideDelay']
            }], message: [{
                type: Input,
                args: ['dscTooltip']
            }] } });
