// sidsc-components/esm2022/dsc-tooltip/dsc-tooltip.module.mjs
import { MAT_TOOLTIP_SCROLL_STRATEGY_FACTORY_PROVIDER } from '@angular/material/tooltip';
import { NgModule } from '@angular/core';
import { AsyncPipe, NgClass } from '@angular/common';
import { DscTooltipDirective } from './dsc-tooltip.directive';
import { DscTooltipComponent } from './dsc-tooltip.component';
import * as i0 from "@angular/core";
export class DscTooltipModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscTooltipModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.2.12", ngImport: i0, type: DscTooltipModule, declarations: [DscTooltipDirective, DscTooltipComponent], imports: [NgClass, AsyncPipe], exports: [DscTooltipDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscTooltipModule, providers: [MAT_TOOLTIP_SCROLL_STRATEGY_FACTORY_PROVIDER] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: DscTooltipModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DscTooltipDirective, DscTooltipComponent],
                    imports: [NgClass, AsyncPipe],
                    exports: [DscTooltipDirective],
                    providers: [MAT_TOOLTIP_SCROLL_STRATEGY_FACTORY_PROVIDER]
                }]
        }] });
