// sidsc-components/esm2022/core/validators/dsc-validators.mjs
import { isEmpty, isNotEmpty } from '../utils/string.util';
import { isEmptyArray, isNotEmptyArray } from '../utils/array.util';
import { FileUploadTypes } from './file-upload-files.enum';
import { formatDataStorageUnits } from '../utils/common.util';
export class DscValidators {
    static min(min) {
        return minValidator(min);
    }
    static max(max) {
        return maxValidator(max);
    }
    static maxTotalFileSize(maxTotalSize) {
        return maxTotalFileSizeValidator(maxTotalSize);
    }
}
export function minValidator(min) {
    return (control) => {
        if (isEmpty(control.value) || isNaN(min))
            return null;
        const value = parseFloat(control.value);
        return !isNaN(value) && value < min
            ? { 'min': { 'min': min, 'actual': control.value } }
            : null;
    };
}
export function maxValidator(max) {
    return (control) => {
        if (isEmpty(control.value) || isNaN(max))
            return null;
        const value = parseFloat(control.value);
        return !isNaN(value) && value > max
            ? { 'max': { 'max': max, 'actual': control.value } }
            : null;
    };
}
export function acceptValidator(accept) {
    return (control) => {
        if (!accept)
            return null;
        const value = control.value;
        const accepts = acceptResolver(accept);
        if (Array.isArray(value)) {
            return handleFileArray(value, accepts);
        }
        else {
            return handleSingleFile(value, accepts);
        }
    };
}
function handleFileArray(files, accepts) {
    if (isEmptyArray(files))
        return null;
    const notAllowedFiles = files.map(file => checkFileTypes(file, accepts)).filter(Boolean);
    return notAllowedFiles.length > 0 ? { 'notAllowedFile': notAllowedFiles } : null;
}
function handleSingleFile(file, accepts) {
    if (!file)
        return null;
    const notAllowedFile = checkFileTypes(file, accepts);
    return notAllowedFile ? { 'notAllowedFile': notAllowedFile } : null;
}
function acceptResolver(accept) {
    const values = accept.replace(/\s/g, '').split(',');
    return isNotEmptyArray(values) ? values.slice() : [];
}
function checkFileTypes(file, accepts) {
    if (!file)
        return null;
    const FILE_EXT_REG = /(^[.]\w*)$/m;
    // @ts-ignore
    const fileExtension = file.name.split('.').pop().toLowerCase();
    const fileType = getFileType(file, fileExtension);
    // @ts-ignore
    const isFound = accepts.some(type => FILE_EXT_REG.test(type) ? type === `.${fileExtension}` : new RegExp(type).test(fileType));
    return isFound ? null : { allowedTypes: accepts, actual: fileType, file };
}
function getFileType(file, fileExtension) {
    const type = file.type;
    if (isNotEmpty(type))
        return type;
    // @ts-ignore
    return FileUploadTypes[fileExtension];
}
export function maxTotalFileSizeValidator(maxTotalSize) {
    return (control) => {
        const value = control.value;
        let sum = 0;
        if (Array.isArray(value)) {
            if (isEmptyArray(value))
                return null;
            sum = value.map(file => file.size).reduce((a, b) => a + b, 0);
        }
        else {
            if (!value)
                return null;
            sum = value.size;
        }
        // @ts-ignore
        const toLargeFile = checkFileSize(sum, maxTotalSize);
        return toLargeFile ? { 'fileSizeOverLimit': toLargeFile } : null;
    };
}
function checkFileSize(actualSize, maxSize) {
    return (Number.isInteger(maxSize) && actualSize > maxSize)
        ? { maxSize, actualSize, actualSizeFriendly: formatDataStorageUnits(actualSize), maxSizeFriendly: formatDataStorageUnits(maxSize) }
        : null;
}
