// sidsc-components/esm2022/dsc-table/shared/dsc-table-column-builder.mjs
import { coerceNumberProperty } from '@angular/cdk/coercion';
import { DscTableColumn } from './dsc-table-column';
export class DscTableColumnBuilder {
    constructor() {
        this._headerAlign = 'left';
        this._bodyCellAlign = 'left';
        this._sortable = false;
        this._sortableActive = false;
        this._showUnsortedIcon = true;
        this._currentDirection = '';
        this._headerFixed = false;
    }
    static instance() {
        return new DscTableColumnBuilder();
    }
    title(title) {
        this._title = title;
        return this;
    }
    property(property) {
        this._property = property;
        return this;
    }
    value(value) {
        this._value = value;
        return this;
    }
    width(width) {
        this._width = coerceNumberProperty(width);
        return this;
    }
    headerAlign(headerAlign) {
        this._headerAlign = headerAlign;
        return this;
    }
    bodyCellAlign(bodyCellAlign) {
        this._bodyCellAlign = bodyCellAlign;
        return this;
    }
    sortable(sortable) {
        this._sortable = sortable;
        return this;
    }
    sortID(sortID) {
        this._sortID = sortID;
        return this;
    }
    sortableActive(sortableActive) {
        this._sortableActive = sortableActive;
        return this;
    }
    showUnsortedIcon(showUnsortedIcon) {
        this._showUnsortedIcon = showUnsortedIcon;
        return this;
    }
    currentDirection(currentDirection) {
        this._currentDirection = currentDirection;
        return this;
    }
    footerTitle(footerTitle) {
        this._footerTitle = footerTitle;
        return this;
    }
    footerCalc(footerCalc) {
        this._footerCalc = footerCalc;
        return this;
    }
    fixed(fixed) {
        this._fixed = fixed;
        return this;
    }
    headerFixed(headerFixed) {
        this._headerFixed = headerFixed;
        return this;
    }
    headerTooltip(headerTooltip) {
        this._headerTooltip = headerTooltip;
        return this;
    }
    headerTooltipVariant(headerTooltipVariant) {
        this._headerTooltipVariant = headerTooltipVariant;
        return this;
    }
    template(template) {
        this._template = template;
        return this;
    }
    build() {
        return new DscTableColumn(this._title, this._property, this._value, this._width, this._headerAlign, this._bodyCellAlign, this._sortable, this._sortID, this._sortableActive, this._showUnsortedIcon, this._currentDirection, this._footerTitle, this._footerCalc, this._fixed, this._headerFixed, this._headerTooltip, this._headerTooltipVariant, this._template);
    }
}
