// sidsc-components/esm2022/dsc-table/shared/dsc-table-column.mjs
export class DscTableColumn {
    constructor(title, property, value, width, headerAlign, bodyCellAlign, sortable, sortID, sortableActive, showUnsortedIcon, currentDirection, footerTitle, footerCalc, fixed, headerFixed, headerTooltip, headerTooltipVariant, template) {
        this.title = title;
        this.property = property;
        this.value = value;
        this.width = width;
        this.headerAlign = headerAlign;
        this.bodyCellAlign = bodyCellAlign;
        this.sortable = sortable;
        this.sortID = sortID;
        this.sortableActive = sortableActive;
        this.showUnsortedIcon = showUnsortedIcon;
        this.currentDirection = currentDirection;
        this.footerTitle = footerTitle;
        this.footerCalc = footerCalc;
        this.fixed = fixed;
        this.headerFixed = headerFixed;
        this.headerTooltip = headerTooltip;
        this.headerTooltipVariant = headerTooltipVariant;
        this.template = template;
        this._sortable = false;
        this._sortableActive = false;
        this._showUnsortedIcon = true;
        this._currentDirection = '';
        this._title = title;
        this._property = property;
        this._value = value;
        this._width = width;
        this._headerAlign = headerAlign;
        this._bodyCellAlign = bodyCellAlign;
        this._sortable = sortable;
        this._sortID = sortID;
        this._sortableActive = sortableActive;
        this._showUnsortedIcon = showUnsortedIcon;
        this._currentDirection = currentDirection;
        this._footerTitle = footerTitle;
        this._footerCalc = footerCalc;
        this._fixed = fixed;
        this._headerFixed = headerFixed;
        this._headerTooltip = headerTooltip;
        this._headerTooltipVariant = headerTooltipVariant;
        this._template = template;
    }
}
